funnel/models/event.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+from . import db, BaseMixin
+from .space import ProposalSpace
+from .user import User
+import random
+import uuid
+
+PRINTABLE_ASCII = map(chr, range(32, 127))
+
+
+def make_key():
+    key = str(uuid.uuid4()).replace('-', '')
+    return ''.join(random.sample(key, len(key)))
+
+
+def make_public_key():
+    return make_key()[:8]
+
+
+def make_private_key():
+    return make_key()[:8]
+
+
+__all__ = ['Event', 'TicketType', 'EventTicketType', 'Participant', 'Attendee', 'SyncTicket']
+
+ATTENDEE_TABLE_NAME = 'attendee'
+
+
+class EventTicketType(BaseMixin, db.Model):
+    """ Join Model for Event and TicketType
+    """
+    __tablename__ = 'event_ticket_type'
+
+    id = db.Column(db.Integer, primary_key=True, unique=True)
+    event_id = db.Column(None, db.ForeignKey('event.id'), nullable=False)
+    ticket_type_id = db.Column(None, db.ForeignKey('ticket_type.id'), nullable=False)
```

Comment:
In this case you have a join table with a composite primary key. To use the secondary declaration properly, you must not define this as a class or you'll have database errors. You have to make a table instead using `db.Table`.